csrc/multidevice/communication.cpp

```diff
@@ -378,31 +378,47 @@ c10::intrusive_ptr<c10d::Work> postScatter(
     c10d::Backend* backend,
     at::Tensor input_tensor,
     at::Tensor output_tensor) {
-  if (my_device_index == communication->root() &&
-      !communication->out()->getDeviceMesh().has(communication->root())) {
-    output_tensor = at::empty_like(input_tensor.slice(0, 0, 1));
-  }
-  std::vector<at::Tensor> output_tensors({output_tensor});
+  NVF_ERROR(
+      isTvContiguous(communication->in()), "Input tensor is not contiguous");
+  NVF_ERROR(
+      isTvContiguous(communication->out()), "Output tensor is not contiguous");
 
+  auto output_device_mesh = communication->out()->getDeviceMesh();
+  bool output_has_root = output_device_mesh.has(communication->root());
```

Comment:
@samnordmann , I vaguely remember at some point you wanted to stop supporting !out_has_root, which could simplify the code here (and probably other rooted communications) a lot. Is that still a valid plan even though it's not your best interest to do that right now?

Reply:
Yes, I think that is a reasonable simplification. I first added these non-orthodox scenarios for completeness, but
1) how we handle them is not ideal (I would prefer handling it during lowering and not at runtime, with zcopy solutions), and,
2) they diverge from MPI standards and makes things more cumbersome for no concrete benefit imo. 

So, as far as I'm concerned, it is ok to remove support for cases where the root is not in the collective.

Reply:
I'll remove this scenario and the related tests then.